cmd/agent/common/misconfig/mounts.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package misconfig
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func procMount() error {
+	groups, err := os.Getgroups()
+	if err != nil {
+		return fmt.Errorf("failed to get process groups: %v", err)
+	}
+	return checkProcMountHidePid("/host/proc/1/mounts", groups)
+}
+
+func checkProcMountHidePid(path string, groups []int) error {
+	file, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "failed to open %s", path)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		if len(fields) != 6 ||
+			fields[0] != "proc" || fields[1] != "/proc" {
```

Comment:
We shouldn’t check the first column (`fields[0] != "proc"`) because the first column is the device to mount and, as `proc` is a pseudo-filesystem, we can put anything as device.

For ex.:
```
root@agent-dev-lenaic:~# ls /tmp/toto/
1    138  149  159  20    2505  3    34    384   4     464   5795  5961  6103  652  795        cmdline    execdomains  kallsyms     loadavg  net           slabinfo       timer_list
10   139  15   16   21    259   30   35    385   40    465   5822  5970  628   663  814        consoles   fb           kcore        locks    pagetypeinfo  softirqs       tty
[…]
root@agent-dev-lenaic:~# grep toto /proc/mounts
anything /tmp/toto proc rw,relatime 0 0
```

On the other hand, we should check the third column (`fields[2]`), because this is the filesystem and here, we want to check the `proc` filesystem.